docs/spring-server/spring-bean-factory.md

```diff
@@ -0,0 +1,49 @@
+---
+id: spring-bean-factory
+title: Spring BeanFactory
+---
+
+You may want to have schema inputs at different levels and if you are using Spring components that expect these inputs to be Autowired you will have to use [BeanFactory](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/BeanFactory.html).
+
+There is an example of how to set this up in [TopLevelBeanFactoryQuery.kt](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/examples/spring/src/main/kotlin/com/expediagroup/graphql/examples/query/TopLevelBeanFactoryQuery.kt) however you can also use the alternate approach of adding the [DataFetchingEnvironment](../writing-schemas/nested-queries.md) to access the arguments in parent nodes.
+
+Both methods have their pros and cons but you may find that the `DataFetchingEnvironment` approach means the Spring components are not directly tied together anymore and can be tested in isolation.
```

Comment:
If you're going to introduce the idea that there are multiple pros and cons, you should really talk about them, and in terms that make it clear to a reader how they should weigh their own needs in light of either choice. (In other words, put it in terms of, "if you're doing X or Y, or Z is important to you, choose method A. If you're worried more about W, or you use U or T, then choose method B. If you like ice cream, then you must use method B.")

If you don't want to go into that, then just lose the "Both methods have their pros and cons, but" part, and leave the rest.

Reply:
I took your advice and just consolidated the docs into a single page for multiple ways you can use nested arguments